### ISSUE TITLE
Change message in 'Choose an Image' on time out

### ISSUE THREAD

**Issue by psonnosp** [Komet/MediaElch#874]
**Is your feature request related to a problem? Please describe.**
Now you get the same message "No images found" in the 'Choose an Image'-dialogue no matter what the reason is that no images are shown.

**Describe the solution you'd like**
Only show the "No images found" when there really are no images found and display "Source timed out" or something like that when ME fails to get a response. Using fanart.tv at the moment is a lottery so would be nice to know if the site is worth a visit or not to download the images manually.

**Describe alternatives you've considered**
Skip handling fanart in ME and do it manually, not that much of a difference in how much time it takes at the moment since you have to double check it anyway if you get no results after waiting a minute or two.

**Additional context**
Think it's a good idea to move #834 and #835 up a bit on the to do list since fanart.tv seem to be dying. Site maintenance is done by a single moderator, none of the admins have been around for six months, the owner even longer, and their servers appear to be on their last breath.

**Comment by bugwelle** [comment]
👍 Error reporting is very important. #831 and #870 were the first steps but there is still much to improve.

**Comment by bugwelle** [comment]
#876 may fix your issue.

**Comment by psonnosp** [comment]
It sadly didn't.
Got the 'No image found' when trying to add a clearart image, went to the site and discovered there was an image available and when I then tried ME once again it appeared.

**Comment by bugwelle** [comment]
Did the image on fanart.tv have at least one vote? Those without aren't provided by the fanart.tv API. 

What movie did you try to download images for? :-)

**Comment by psonnosp** [comment]
That one vote thing doesn't seem to be a constant. Looks like it was just a coincidence that it worked out that way when I created #858, think it's a delay between when images are approved on the site and when they show up in the API for some reason.
But it was the thumb for The Two Popes, has no votes but are listed in the [API](https://webservice.fanart.tv/v3/movies/551332?api_key=842f7a5d1cc7396f142b8dd47c4ba42b) and it came up in ME when I tried the second time so that's not it.
The issue is that it didn't show the first time I tried and got the standard "No images found" when it should have been one telling me that the server timed out or whatever the reason was.

**Comment by bugwelle** [comment]
You are right. Thank you for clearing that up.

**Comment by bugwelle** [comment]
*Note for myself:*

For this issue to be fixed, image providers must be able to report errors. That is as of now not possible, yet. For example

```cpp
connect(provider, SIGNAL(sigImagesLoaded(QVector<Poster>)), this, SLOT(onProviderImagesLoaded(QVector<Poster>)));
```

If the image provider is finished (either successfully or with errors) then `sigImagesLoaded` is emitted. There is no way to report issues here...
Furthermore `sigImagesLoaded` is misleading. Posters aren't downloaded but only their URLs. We later pass them to `setDownloads`.

**Comment by bugwelle** [comment]
Will be fixed by #882
